view/frontend/templates/checkout/onepage/success/promptpay_additional_info.phtml

```diff
@@ -1,25 +1,76 @@
-<p>
-    Payment expires at: <?= $block->escapeHtml($block->getChargeExpiresAt()); ?>
-</p>
-<p>
-    <button class="action secondary" onclick="window.print ? window.print() : alert('Print not available')">
-    <?= $block->escapeHtml(__('Print'))?></button>
-</p>
-
-<div style="padding: 15px; margin: 20px 0; border: 1px solid black">
-    <p><strong>To make payment:</strong></p>
-    <ul style="list-style: auto; margin-bottom: 0px; padding: 0px 15px;">
-        <li>Download the QR code or open your preferred bank app to scan it</li>
-        <li>Check that the payment details are correct</li>
-        <li>Import the QR code image into your bank app or scan the QR code with your bank app to pay</li>
-        <li>Share the payment slip from your bank app to the seller</li>
-    </ul>
+<div id="qr-info" style="margin-bottom: 10px;">
+    <p>
+        <?= $block->escapeHtml(__('Amount to pay')) ?>: 
+        <strong>
+            <?= $block->escapeHtml($block->getOrderAmount()) ?>
+            </b>
+        </strong>
+    <p style="margin:.5rem 0"><?= $block->escapeHtml(__('PromptPay QR Code.')); ?></p>
+    <div><img id="img_payment_code" src="<?= $block->escapeHtml($block->getImageCode()) ?>" /></div>
+    <p>
+        Payment expires in: <span id="countdown"></span>
+    </p>
+    <p>
+        <button class="action secondary" onclick="window.print ? window.print() : alert('Print not available')">
+        <?= $block->escapeHtml(__('Print'))?></button>
+    </p>
+
+    <div style="padding: 15px; margin: 20px 0; border: 1px solid black">
+        <p><strong>To make payment:</strong></p>
+        <ol style="margin-bottom: 0px; padding: 0px 15px;">
+            <li>Download the QR code or open your preferred bank app to scan it</li>
+            <li>Check that the payment details are correct</li>
+            <li>Import the QR code image into your bank app or scan the QR code with your bank app to pay</li>
+            <li>Share the payment slip from your bank app to the seller</li>
+        </ol>
+    </div>
 </div>
+
+<script>
+(function () {
+    let countDownInterval;
+
+    function calculateCountdown() {
+        const currentDateTime = new Date();
+        const expiresAtDateTime = new Date("<?= $block->escapeHtml($block->getChargeExpiresAt()) ?>");
+        const difference = expiresAtDateTime - currentDateTime;
+        const hours = Math.floor((difference % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        const minutes = Math.floor((difference % (1000 * 60 * 60)) / (1000 * 60));
+        const seconds = Math.floor((difference % (1000 * 60)) / 1000);
+        return { hours, minutes, seconds };
+    }
+
+    function padZero(num) {
+        return num.toString().padStart(2, '0');
+    }
+
+    function hideQrInfoAndShowPaymentExpired() {
+        const element = document.getElementById('qr-info')
+        element.innerHTML = 'Payment Expired.'
```

Comment:
Let's include some styling to create an alert message that will be easily noticeable by people.

Reply:
Updated